spec/graphql/union_type_spec.rb

```diff
@@ -14,11 +14,28 @@
     }
   }
 
+  class FilterType2 < GraphQL::Schema::TypeMembership
+    def visible?(ctx)
+      return true if visibility.nil?
+
+      visibility[:private] && ctx[:private]
+    end
+  end
+
+  class FilteredUnion < GraphQL::UnionType 
+    type_visibility_class FilterType2
+
+    possible_types OpenStruct.new(kind: GraphQL::TypeKinds::OBJECT), visibility: { private: true }
```

Comment:
This was not letting me use `type_1` or `type_2` within the class, if there are thoughts on how to fix this let me know :D

Reply:
(I don't think it's possible -- those `let(...)` declarations create methods that are accessible inside the `it do ... end` blocks, but the methods aren't available anyplace else.)